src/plugins/intel_gna/src/transformations/gather_sinking_transpose_reshape.cpp

```diff
@@ -19,154 +19,120 @@
 using namespace ov::intel_gna;
 using namespace ov::intel_gna::pass;
 using namespace ov::intel_gna::limitations;
-using namespace ov::opset9;
+using namespace ov::opset12;
 using namespace ov::pass::pattern;
 using namespace gather_sinking;
 
 namespace {
 
-using NodePtr = std::shared_ptr<ov::Node>;
-using NodePair = std::pair<NodePtr, NodePtr>;
+std::vector<std::shared_ptr<ov::Node>> SinkForward(std::shared_ptr<ov::Node> transpose,
+                                                   std::shared_ptr<ov::Node> reshape) {
+    std::vector<std::shared_ptr<ov::Node>> new_nodes = {};
 
-std::vector<size_t> CreateGatherIndices(const ov::Shape& input_shape, const ov::Shape& order) {
-    if (input_shape.size() < 2 || input_shape.size() > 4) {
-        THROW_GNA_EXCEPTION << "Usupported shape size: " << input_shape.size();
-    }
-
-    ov::Shape input_shape_4d = input_shape;
-    ov::Shape order_4d = order;
-    // Just to simplify the code we transform all shapes to 4d by adding 1 dimentions at the end
-    while (input_shape_4d.size() < 4) {
-        input_shape_4d.push_back(1);
-        order_4d.push_back(order_4d.size());
-    }
-    ov::Shape output_shape_4d = graph_utils::transpose_shape(input_shape_4d, order_4d);
-
-    // common case when shape is 4d
-    std::vector<size_t> xyz_4d = {input_shape_4d[3] * input_shape_4d[2] * input_shape_4d[1],
-                                  input_shape_4d[3] * input_shape_4d[2],
-                                  input_shape_4d[3],
-                                  1};
-
-    std::vector<size_t> xyz = graph_utils::transpose_shape(xyz_4d, order_4d);
-    std::vector<size_t> gather_order;
-
-    for (size_t n = 0; n < output_shape_4d[0]; ++n) {
-        for (size_t i = 0; i < output_shape_4d[1]; ++i) {
-            for (size_t j = 0; j < output_shape_4d[2]; ++j) {
-                for (size_t k = 0; k < output_shape_4d[3]; ++k) {
-                    gather_order.push_back(n * xyz[0] + i * xyz[1] + j * xyz[2] + k * xyz[3]);
-                }
-            }
+    auto transpose_const = ov::as_type_ptr<Constant>(transpose->get_input_node_shared_ptr(1));
+    const auto gather_indices_value =
+        graph_utils::make_gather_indexes_from_transpose_axes(transpose->get_input_shape(0),
```

Comment:
Please use indexes/indices consistently here and in other places.

Reply:
we agreed to use indexes, this is not a new function

Reply:
renamed